spec/controllers/users/edit_password_controller_spec.rb

```diff
@@ -27,6 +27,32 @@
         expect(response).to redirect_to profile_url
         expect(flash[:notice]).to eq t('notices.password_changed')
       end
+
+      it 're-encrypts PII on active profile' do
+        stub_sign_in
+
+        user = controller.current_user
+        profile = create(:profile, :active, user: user, pii: { ssn: '1234' })
+
+        old_password = user.password
+        new_password = 'salty new password'
+
+        cacher = Pii::Cacher.new(user, controller.user_session)
+        cacher.save(old_password, profile)
+
+        email_notifier = instance_double(EmailNotifier)
+        allow(EmailNotifier).to receive(:new).with(user).and_return(email_notifier)
+
+        expect(email_notifier).to receive(:send_password_changed_email)
+
+        params = { password: new_password, current_password: old_password }
+        patch :update, update_user_password_form: params
+
+        profile.reload
+
+        expect(profile.decrypt_pii(new_password)).to be_a Pii::Attributes
```

Comment:
good call, @zachmargolis -- I had actually such a test in there and then took it out because it was failing for exactly the reason you mention.

I will add it back as pending.